### DISCUSSION TITLE
Choose between a floating design or not for URL bar

### DISCUSSION THREAD

**Discussion by utopiaeh** [zen-browser/desktop#3512] [Q&A] [answered]
Add a setting to use the old version of the URL bar, allowing users to choose between a floating design or not.

**Comment by MeguminSama** [comment]
Reasons I want to be able to use the normal URL bar:

- When I click the URL bar, it's no longer highlighted fully. I now have to click the URL bar, then move my mouse to the new location and triple click to fully select the URL.
- It covers more of the webpage, which i may still want to look at while typing in the address box.

**Comment by dustinrouillard** [comment]
Please, please, please. I use Zen for the minimal feel but will move back to firefox if there is not a setting for this, I don't want to entirely change how the web should be used.

**Reply by dustinrouillard** [reply]
If we're gonna be this different with any design choices in the future please make them settings before release, this will be the second time I choose not to update and to disable auto updates because of a change that was forced upon me.

**Comment by dhay3** [comment]


**Reply by MeguminSama** [reply]
You should probably create a separate issue for this, as it's not relevant to the request.

**Comment by Gorgaer** [comment]
This behavior of the URL bar is not very convenient. I would like to have an option to disable the floating design for this.

**Comment by mateuszkozakiewicz** [comment]
While I don't mind the idea the implementation is awful at the moment, with so many issues it should be disabled behind a flag not even a setting because it is unusable
- when you click the URL it is not being selected, you need ctrl+a
- the theme is inconsistent with anything
- the position left/right is not centered at times
- the original URL bar at the top remains as an empty black hole?
- it adds nothing to the functionality
- there was already zen mod for this which did it better
- 
there doesn't seem to be a need for this change at all

**Reply by dustinrouillard** [reply]
This might be the best description of almost everything that is wrong with this "feature".
Please remove as default thx

**Comment by MaxHart828** [comment]
Agreed. Change it back or give us the option. 

The latest (1.0.2-b.1) says a new feature offers to disable this but I've looked and can't find it. If it indeed exists, at least tell us where to find the feature.

**Reply by Muhammad-Mohid-Khan** [reply]
yep they said it but they haven't added it yet very weird

**Reply by natsukagami** [reply, marked as answer]
The new release adds `zen.view.use-deprecated-urlbar` as a preference you can toggle in `about:config`. Setting it to true will restore the old URL bar :D

**Reply by utopiaeh** [reply]
Yes, but in my case, the background for the URL bar is transparent, making it very uncomfortable to use.

**Reply by MeguminSama** [reply]
it's transparent for me as well

**Reply by chenasraf** [reply]
Same

**Reply by dantaeusb** [reply]
Well, yes, it's quite unusable in the current state.

It's also just a little bit misaligned, but that's not a big deal – a few pixels is not equivalent to a half a screen for the eyes to travel that I would have to do with a default setting.

<img width="1112" alt="Screenshot 2024-12-12 at 22 53 31" src="https://github.com/user-attachments/assets/d02ae6d5-3309-4565-8d6b-feb1a1cf4143" />

**Reply by dustinrouillard** [reply]
Even a config flag doesn't sound like a good option, especially when you call it "deprecated", please make it an official setting, this currently makes me think its going to be neglected and eventually broken or even removed, and I'm sure I'm not the only one who thinks this.

**Reply by DutchPete** [reply]
> The new release adds `zen.view.use-deprecated-urlbar` as a preference you can toggle in `about:config`. Setting it to true will restore the old URL bar :D

This is my URL bar: floating when CTRL+L is typed:

![image](https://github.com/user-attachments/assets/fb17dd84-8285-4901-8e05-44c38b4206a2)

which is the same as before adjusting the about:config rule per your instruction.
Is it possible to have the YURL bar permanently?

**Reply by egl1** [reply]
If you change the Zen URL bar setting in settings to normal it will always appear on the left for the compact view and on the top if you have the url bar there

**Reply by DutchPete** [reply]
> If you change the Zen URL bar setting in settings to normal it will always appear on the left for the compact view and on the top if you have the url bar there

I had that setting set to "normal", but it did not change anything.
In the Discord channel someone suggested to set `zen.urlbar.replace-newtab` to "False" in about:config. That did the trick.

**Reply by DutchPete** [reply]
> The new release adds zen.view.use-deprecated-urlbar as a preference you can toggle in about:config. Setting it to true will restore the old URL bar :D

Rather than use a deprecated setting, there is the about:config setting `zen.urlbar.replace-newtab` that can be set to "False" and does the trick too.

**Comment by chenasraf** [comment]
It also messes up with the ability to easily update the URL. I sometimes need (as a dev) to remove, move or add words in the URL bar, and I usually would double click the word for this, immediately selecting it for typing stuff on it or cutting it. Now, I have to click once, move the mouse again, THEN double click. The `about:config` fix leaves the dropdown transparent.

**Reply by Muhammad-Mohid-Khan** [reply]
also for example if i type 'y" for youtube and click on it it doesn't select the first time ALWAYS and just leaves the y i have to go again and then do it again
